internal/pkg/service/appsproxy/proxy/proxy_test.go

```diff
@@ -558,32 +570,34 @@ func TestAppProxyRouter(t *testing.T) {
 				response, err := client.Do(request)
 				require.NoError(t, err)
 				require.Equal(t, http.StatusFound, response.StatusCode)
-				location := response.Header["Location"][0]
+				location := response.Header.Get("Location")
+				assert.Contains(t, location, "oidc/authorize?client_id=")
 				cookies := response.Cookies()
-				assert.Len(t, cookies, 2)
-
-				assert.Equal(t, "_oauth2_provider", cookies[0].Name)
-				assert.Equal(t, "oidc", cookies[0].Value)
-				assert.Equal(t, "/", cookies[0].Path)
-				assert.Equal(t, "oidc.hub.keboola.local", cookies[0].Domain)
-				assert.True(t, cookies[0].HttpOnly)
-				assert.True(t, cookies[0].Secure)
-				assert.Equal(t, http.SameSiteLaxMode, cookies[0].SameSite)
-
-				assert.Equal(t, "_oauth2_proxy_csrf", cookies[1].Name)
-				assert.Equal(t, "/", cookies[1].Path)
-				assert.Equal(t, "oidc.hub.keboola.local", cookies[1].Domain)
-				assert.True(t, cookies[1].HttpOnly)
-				assert.True(t, cookies[1].Secure)
-				assert.Equal(t, http.SameSiteNoneMode, cookies[1].SameSite)
+				if assert.Len(t, cookies, 2) {
+					assert.Equal(t, "_oauth2_provider", cookies[0].Name)
+					assert.Equal(t, "oidc", cookies[0].Value)
+					assert.Equal(t, "/", cookies[0].Path)
+					assert.Equal(t, "oidc.hub.keboola.local", cookies[0].Domain)
+					assert.True(t, cookies[0].HttpOnly)
+					assert.True(t, cookies[0].Secure)
+					assert.Equal(t, http.SameSiteLaxMode, cookies[0].SameSite)
+
+					assert.Equal(t, "_oauth2_proxy_csrf", cookies[1].Name)
+					assert.Equal(t, "/", cookies[1].Path)
+					assert.Equal(t, "oidc.hub.keboola.local", cookies[1].Domain)
+					assert.True(t, cookies[1].HttpOnly)
+					assert.True(t, cookies[1].Secure)
+					assert.Equal(t, http.SameSiteNoneMode, cookies[1].SameSite)
+				}
```

Comment:
same as above ^ just adjust the values, not only providers. Or create cookies type object so you deep equal the fields? This does not look good. Reflection also solves possible changes in `cookie` structure etc.